themes-default/slim/src/components/subtitle-search.vue

```diff
@@ -0,0 +1,258 @@
+                            codes: response.data.subtitles,
+                            languages: response.data.languages
+                        });
+                    }
+                })
+                .catch(error => {
+                    console.log(`Error trying to search for subtitles. Error: ${error}`);
+                })
+                .finally(() => {
+                    // Destroy this component.
+                    this.loadingMessage = '';
+                    this.loading = false;
+                    this.$destroy();
+                });
+        },
+        manualSearch() {
+            const { show, season, episode } = this;
+
+            this.displayQuestion = false;
+            this.loading = true;
+            this.loadingMessage = 'Searching for subtitles... ';
+            const url = `home/manualSearchSubtitles?indexername=${show.indexer}&seriesid=${show.id[show.indexer]}&season=${season}&episode=${episode}`;
+            apiRoute(url) // eslint-disable-line no-undef
+                .then(response => {
+                    if (response.data.result === 'success') {
+                        this.subtitles.push(...response.data.subtitles);
+                    }
+                }).catch(error => {
+                    console.log(`Error trying to search for subtitles. Error: ${error}`);
+                    this.$destroy();
+                }).finally(() => {
+                    this.loading = false;
+                });
+        },
+        pickSubtitle(subtitleId) {
+            // Download and save this subtitle with the episode.
+            const { show, season, episode } = this;
+
+            this.displayQuestion = false;
+            this.loadingMessage = 'downloading subtitle... ';
+            this.loading = true;
+            const url = `home/manualSearchSubtitles?indexername=${show.indexer}&seriesid=${show.id[show.indexer]}&season=${season}&episode=${episode}&picked_id=${subtitleId}`;
+            apiRoute(url) // eslint-disable-line no-undef
+                .then(response => {
+                    if (response.data.result === 'success') {
+                        // Update the show, as we have new information (subtitles)
+                        // Let's emit an event, telling the displayShow component, to update the show using the api/store.
+                        this.$emit('update', {
+                            reason: 'new subtitles found',
+                            codes: response.data.subtitles,
+                            languages: response.data.languages
+                        });
+                    }
+                })
+                .catch(error => {
+                    console.log(`Error trying to search for subtitles. Error: ${error}`);
+                })
+                .finally(() => {
+                    // Destroy this component.
+                    this.loadingMessage = '';
+                    this.loading = false;
+                    this.$destroy();
```

Comment:
As I explained above in the `destroyed` section, I don't think you should call `$destroy` on an element.
Plus, the docs say it's not preferable.
https://vuejs.org/v2/api/#vm-destroy
> In normal use cases you shouldn’t have to call this method yourself. Prefer controlling the lifecycle of child components in a data-driven fashion using `v-if` and `v-for`.

Reply:
Except it's used as a dynamic mounted component. So it's setup at runtime within a specific table (episode) row. I need to destroy the component. But i'll try if a bare `this.$destroy` is sufficient.

Reply:
A little bit of context. This is how i'm creating the component in display-show.vue
```
const { id, indexer, getEpisodes, show, subtitleSearchComponents } = this;
            const SubtitleSearchClass = Vue.extend(SubtitleSearch);  // eslint-disable-line no-undef
            const instance = new SubtitleSearchClass({
                propsData: { show, season, episode, key: rowIndex },
                parent: this
            });

            // Update the show, as we downloaded new subtitle(s)
            instance.$on('update', event => {
                // This could be replaced by the generic websocket updates in future.
                if (event.reason === 'new subtitles found') {
                    getEpisodes({ id, indexer, season });
                };
            });

            
        // This was just added, to try to move the destroying to the calling component. That could provide for some more options when using the components. But else I don't see much difference compared to destroying it from within the component.
        instance.$on('close', vm => {
                // Close the component.
                vm.$destroy();
            });

            const node = document.createElement('div');
            this.$refs['table-seasons'].$refs[`row-${rowIndex}`][0].after(node);
            instance.$mount(node);
            subtitleSearchComponents.push(instance);
        },
```

Reply:
What I still don't understand is, why do you need to create the component like that?

Reply:
![image](https://user-images.githubusercontent.com/1331394/58658791-a1030580-8321-11e9-87a2-54f8cbd2d875.png)
because I only want to create the component after I click on the subtitle button, in the episode row. If a show as 400 episodes. I don't want to create 400 subtitle modals, if I might not even need them. Now I create it, and it will slide in the table.

Reply:
A simple `v-if` on the component would prevent that. `v-if` only creates / renders the component when the expression in it evaluates to true.
If I understand you correctly, then all you need to have is (on the parent component), something like an array that holds episode slugs, that you add a slug of an episode the user clicked on the subtitle search.... 
Eh. I'm tired right now but maybe a code will do better:
```js
data() {
    return {
        subtitleSearchVisible: []
    };
},
methods: {
    closeSubtitleSearch(slug) {
        const index = this.subtitleSearchVisible.indexOf(episode.indentifier);
        if (index > -1) {
            this.$delete(this.subtitleSearchVisible, index);
        }
    }
}
```
```vue
<!-- episode row (inside a v-for loop) -->
<a @click="searchSubtitles(episode.identifier)"><img ... /></a>
<subtitle-search v-if="subtitleSearchVisible.includes(episode.identifier)" @close="closeSubtitleSearch(episode.identifier)"></subtitle-search>
```

Reply:
I'm pretty much done for now. I'll try that in the coming days

Reply:
this:
```
<!-- episode row (inside a v-for loop) -->
<a @click="searchSubtitles(episode.identifier)"><img ... /></a>
<subtitle-search v-if="subtitleSearchVisible.includes(episode.identifier)" @close="closeSubtitleSearch(episode.identifier)"></subtitle-search>
```
is not possible. As i'm using vue-good-table. We don't have a v-for loop.
And i'm currently injecting a <tr></tr> after the episodes row.

Any other ideas?